File: CS/WindowsApplication1/CustomEditor/RepositoryItemMyButtonEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Controls;

namespace WindowsApplication1
{
    [UserRepositoryItem("Register")]

    [System.ComponentModel.DesignerCategory("")]
    public class RepositoryItemMyButtonEdit : RepositoryItemButtonEdit
    {
                private RepositoryItemRichTextEdit _RichRI = new RepositoryItemRichTextEdit();
        public RepositoryItemRichTextEdit RichEdit
        {
            get { return _RichRI; }
            set { _RichRI = value; }
        }

        static RepositoryItemMyButtonEdit()
        {
            Register();
        }
        public RepositoryItemMyButtonEdit() { }

        internal const string EditorName = "MyButtonEdit";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyButtonEdit),
                typeof(RepositoryItemMyButtonEdit), typeof(MyButtonEditViewInfo),
                new MyPainter(), true, null));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }

        public override TextEditStyles TextEditStyle
        {
            get
            {
                return TextEditStyles.DisableTextEditor;
            }
            set
            {

            }
        }

        public override void Assign(RepositoryItem item)
        {
            base.Assign(item);
            RichEdit.Assign((item as RepositoryItemMyButtonEdit).RichEdit);
        }


    }
}

File: VB/WindowsApplication1/CustomEditor/RepositoryItemMyButtonEdit.vb
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Controls

Namespace WindowsApplication1

    <UserRepositoryItem("Register")>
    <System.ComponentModel.DesignerCategory("")>
    Public Class RepositoryItemMyButtonEdit
        Inherits RepositoryItemButtonEdit

        Private _RichRI As RepositoryItemRichTextEdit = New RepositoryItemRichTextEdit()

        Public Property RichEdit As RepositoryItemRichTextEdit
            Get
                Return _RichRI
            End Get

            Set(ByVal value As RepositoryItemRichTextEdit)
                _RichRI = value
            End Set
        End Property

        Shared Sub New()
            Call Register()
        End Sub

        Public Sub New()
        End Sub

        Friend Const EditorName As String = "MyButtonEdit"

        Public Shared Sub Register()
            Call EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyButtonEdit), GetType(RepositoryItemMyButtonEdit), GetType(MyButtonEditViewInfo), New MyPainter(), True, CType(Nothing, System.Drawing.Image)))
        End Sub

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return EditorName
            End Get
        End Property

        Public Overrides Property TextEditStyle As TextEditStyles
            Get
                Return TextEditStyles.DisableTextEditor
            End Get

            Set(ByVal value As TextEditStyles)
            End Set
        End Property

        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            MyBase.Assign(item)
            RichEdit.Assign(TryCast(item, RepositoryItemMyButtonEdit).RichEdit)
        End Sub
    End Class
End Namespace
